backend/app/rag/embeddings.py

```diff
@@ -28,10 +29,16 @@ def get_embedding_model(model_type: str = 'gemini') -> Embeddings:
         ValueError: If an unsupported model type is specified.
     """
     if model_type == 'gemini':
+        if not settings.GEMINI_API_KEY:
+            print('No GEMINI_API_KEY to create embedding model')
+            return None
```

Comment:
Shouldn't we raise an error here as well? Because the code fill fail anyways at some point if no API key is provided.
Then can you can also change the method signature to not have `None`